st2common/st2common/transport/bootstrap_utils.py

```diff
@@ -27,16 +27,38 @@
 from st2common.transport.reactor import SENSOR_CUD_XCHG
 from st2common.transport.reactor import TRIGGER_CUD_XCHG, TRIGGER_INSTANCE_XCHG
 
+from st2common.transport import actionexecutionstate
+from st2common.transport import announcement
+from st2common.transport import execution
+from st2common.transport import liveaction
+from st2common.transport import reactor
+
 LOG = logging.getLogger('st2common.transport.bootstrap')
 
 __all__ = [
     'register_exchanges'
 ]
 
+# List of exchanges which are pre-declared on service set up.
 EXCHANGES = [ACTIONEXECUTIONSTATE_XCHG, ANNOUNCEMENT_XCHG, EXECUTION_XCHG, LIVEACTION_XCHG,
              LIVEACTION_STATUS_MGMT_XCHG, TRIGGER_CUD_XCHG, TRIGGER_INSTANCE_XCHG,
              SENSOR_CUD_XCHG]
 
+# List of queues which are pre-declared on service set up.
+# Because of the worker model used, this is required with some non-standard transports such as
+# Redis one. Even though kombu requires queues to be pre-declared upfront in such scenarios,
+# RabbitMQ transport doesn't require that.
+QUEUES = [
```

Comment:
Just checked.

Sadly, https://github.com/StackStorm/st2/issues/3290 MQ-related bug is still clearly reproducible with this new `predeclare_queues=True`